cmd/gateway-azure.go

```diff
@@ -40,6 +40,19 @@ const globalAzureAPIVersion = "2016-05-31"
 const azureBlockSize = 100 * humanize.MiByte
 const metadataObjectNameTemplate = globalMinioSysTmp + "multipart/v1/%s.%x/azure.json"
 
+// Tests if s3Metadata has double "-", we cannot handle this
+// without introducing a new encoding style.
+func isValidAzureMetadata(s3Metadata map[string]string) (match bool) {
```

Comment:
The name is inappropriate - what is meant is `hasUnsupportedAzureMetadata`.

Reply:
ah correct will fix.